tools/HeII_input_file_maker.py

```diff
@@ -0,0 +1,280 @@
+        dataarr[1,:] = [min(x[i,0], 1) for i in range(numz)]
+        return dataarr
+
+    def _makexHeIIInterp(self):
+        """Produces outfile where columns are z, xHeIII, and number of ionizing photons per nHe produced. Returns an interpolation function."""
+        dataarr = self.xHeIII_quasar(self.z_f, self.z_i)
+        return scipy.interpolate.interp1d(dataarr[0,:], dataarr[1,:], bounds_error=False, fill_value=0.0)
+
+    def quasar_emissivity_HM12(self, redshift):
+        """Proper emissivity of HeII ionizing photons from Haardt & Madau (2012) (1105.2039.pdf eqn 37)"""
+        enhance_fac=1
+        epsilon_nu = enhance_fac*3.98e24*(1+redshift)**7.68*np.exp(-0.28*redshift)/(np.exp(1.77*redshift) + 26.3) #erg s^-1 MPc^-3 Hz^-1
+        e = epsilon_nu/(self.h_erg_s*self.alpha_q)/(self.mpctocm**3)*4.**(-self.alpha_q)
+        return e
+
+    def quasar_emissivity_K15(self, redshift):
+        """Proper emissivity of HeII ionizing photons from Khaire + (2015)"""
+        epsilon_nu = 10.**(24.6)*(1.+redshift)**8.9 * np.exp(-0.36*redshift)/(np.exp(2.2*redshift)+25.1)  #erg s^-1 MPc^-3 Hz^-1
+        e = epsilon_nu/(self.h_erg_s*self.alpha_q)/(self.mpctocm**3)*4.**(-self.alpha_q)
+        return e
+
+    def quasar_emissivity_Kulkarni19(self, redshift):
+        """Proper emissivity of HeII ionizing photons from Kulkarni + (2019)"""
+        epsilon_nu = 10.**(24.72)*(1.+redshift)**8.42 * np.exp(-2.1*redshift)/(np.exp(1.09*redshift)+38.56)  #erg s^-1 MPc^-3 Hz^-1
+        e = epsilon_nu/(self.h_erg_s*self.alpha_q)/(self.mpctocm**3)*4.**(-self.alpha_q)
+        return e
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--alphaq', type=float, default=1.7, help='QSO spectral index', required=True)
+    parser.add_argument('--Emax', type=float, default=150, help='Threshold long-mean-free-path photon energy in eV', required=False)
+    parser.add_argument('--cf', type=float, default=3., help='Subgrid clumping factor', required=False)
+    parser.add_argument('--z_i', type=float, default=-1, help='Start redshift of helium reionization', required=False)
+    parser.add_argument('--z_f', type=float, default=-1, help='End redshift of helium reionization', required=False)
+    parser.add_argument('--hist', type=str, default="quasar", help='Type of reionization history', required=True)
+    parser.add_argument('--outfile', type=str, default="HeIIReionizationTable", help='Name of file to save to', required=False)
+    args = parser.parse_args()
+
+    if args.z_i < 0:
+        if args.hist == "linear":
+            args.z_i = 4.0
+        else:
+            args.z_i = 6.0
+    if args.z_f < 0:
+        if args.hist == "linear":
+            args.z_f = 2.8
+        else:
+            args.z_f = 2.
+
+    heat = HeIIheating(hist = args.hist, z_i = args.z_i, z_f= args.z_f, Emax=args.Emax, alpha_q = args.alphaq, clumping_fac = args.cf)
+    heat.setUpInterpTable(args.outfile)
```

Comment:
WriteInterpTable? the spelling setUp does not suggest writing, making this line a bit mysterious.